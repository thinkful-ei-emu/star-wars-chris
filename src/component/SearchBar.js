import React from 'react'

class SearchBar extends React.Component {
    state = {
        name: '',
        category: 'films/',
        error: null,
    }

    createSearch = (name) => {
        this.setState({ name })
    }

    createCategory = (category) => {
        this.setState({ category })
    }

    handleSubmit = e => {
        e.preventDefault();
        let { name, category } = this.state;

        this.setState({error:null});

        const url = `https://swapi.co/api/${category}?search=${name}`;
        fetch(url)
            .then(res => {
                if(!res.ok) {
                    return res.json().then(error => {
                        throw error;
                    })
                }
                return res.json()
            })
            .then(data => {
                name = '';
                this.props.addResults(data.results);
            })
            .catch(error => {
                this.setState({ error })
            })
    }

    render() {
        return (
            <div>
                <form className="charsearch-form" onSubmit={this.handleSubmit}>
                    <div>
                        <label htmlFor="category">category</label>
                        <select name="category" valule={this.state.category}
                            onChange={e => this.createCategory(e.target.value)}>
                            <option value="films/">Films</option>
                            <option value="people/">People</option>
                            <option value="planets/">Planets</option>
                            <option value="species/">Species</option>
                            <option value="starships/">Starships</option>
                            <option value="vehicles/">Vehicles</option>
                        </select>
                        <label htmlFor="namesearch">Search</label>
                        <input type="text" name="namesearch" 
                            placehoder="Skywalker" value={this.state.name} 
                            onChange={e => this.createSearch(e.target.value)} />
                        <button type="submit">Search</button>
                    </div>
                </form>
                <div>
                    { this.state.error && <p>Your search returned no results. Please try again.</p>}
                </div>
            </div>
        )
    }
}

export default SearchBar;