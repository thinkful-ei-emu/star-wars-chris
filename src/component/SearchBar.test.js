// make React available
import React from 'react';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 


//make the ReactDOM available, necessary for rendering the component
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

//make the App component available
import SearchBar from './SearchBar';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchBar component', () => {
    //this is the test case
    it('renders without crashing', () => {
    // first create a DOM element to render the component into
        const div = document.createElement('div');

        //render the component, this is the actual test, if something is wrong it will fail here
        ReactDOM.render(<SearchBar addResults={(e) => this.addResults(e)} />, div);

        //clean up code
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<SearchBar addResults={(e) => this.addResults(e)} />)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });

    it('calls addResults function when form is submitted', () => {
        const addResultsFn = jest.fn()
        const wrapper = mount(<SearchBar addResults={addResultsFn} />);

        wrapper.find('form').simulate('submit')
        expect(addResultsFn).toHaveBeenCalledTimes(1)
    })
});