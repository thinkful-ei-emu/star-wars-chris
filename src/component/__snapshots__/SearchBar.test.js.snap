// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders the UI as expected 1`] = `
<div>
  <form
    className="charsearch-form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="category"
      >
        category
      </label>
      <select
        name="category"
        onChange={[Function]}
        valule="films/"
      >
        <option
          value="films/"
        >
          Films
        </option>
        <option
          value="people/"
        >
          People
        </option>
        <option
          value="planets/"
        >
          Planets
        </option>
        <option
          value="species/"
        >
          Species
        </option>
        <option
          value="starships/"
        >
          Starships
        </option>
        <option
          value="vehicles/"
        >
          Vehicles
        </option>
      </select>
      <label
        htmlFor="namesearch"
      >
        Search
      </label>
      <input
        name="namesearch"
        onChange={[Function]}
        placehoder="Skywalker"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
  <div />
</div>
`;
